common/logging_extra.py

```diff
@@ -163,6 +163,14 @@ def event(self, event_name, *args, **kwargs):
     else:
       self.info(evt)
 
+  def blocking_info(self, msg, timeout=10):
```

Comment:
Can you explain why this is blocking?

Reply:
My understanding is that this spawns a new thread, starts it, and the join function blocks the calling thread until the target function of thread t finishes.